### PR TITLE
Add test on `UnifiedTestRunner::setEntityMapObserver()`

### PR DESCRIPTION
**Pull request by GromNaN**
`UnifiedTestRunner::setEntityMapObserver()` is used in `[drivers-atlas-testing](https://github.com/mongodb-labs/drivers-atlas-testing/).

### CODE REVIEW

File: tests/UnifiedSpecTests/UnifiedTestRunner.php [hunk 1]
```diff
@@ -113,31 +110,10 @@ public function run(UnifiedTestCase $test): void
...
     }
 
-    /**
-     * Defines a callable to receive the EntityMap after each test.
-     *
-     * This function is primarily used by the Atlas testing workload executor.
-     *
-     * @param callable(EntityMap):void $entityMapObserver
-     */
-    public function setEntityMapObserver(callable $entityMapObserver): void
```
Comment (jmikola) [inline tests/UnifiedSpecTests/UnifiedTestRunner.php:136]: This is called from the PHP workload executor in the Atlas testing project (see: [`workload-executor.php`](https://github.com/mongodb-labs/drivers-atlas-testing/blob/8cc57d7efd954d6133dc7afac5a19e85fe43a873/integrations/php/workload-executor.php#L86)).

If you'd like to repurpose this PR you can add a link to the https://github.com/mongodb-labs/drivers-atlas-testing/ repository.

Comment (GromNaN) [reply tests/UnifiedSpecTests/UnifiedTestRunner.php:131]: Indeed, it was written in the comment above.

I updated the PR to add link and test this feature.

File: tests/UnifiedSpecTests/UnifiedTestRunnerTest.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+namespace MongoDB\Tests\UnifiedSpecTests;
+
+use MongoDB\Tests\FunctionalTestCase;
+
+class UnifiedTestRunnerTest extends FunctionalTestCase
+{
+    public function testEntityMapObserver(): void
+    {
+        $test = UnifiedTestCase::fromFile(__DIR__ . '/crud/find.json');
```
Comment (alcaeus) [inline tests/UnifiedSpecTests/UnifiedTestRunnerTest.php:11]: For the sake of completeness, this test might start failing if the crud tests change significantly. Creating a separate test would be safer, but for the sake of getting this method tested (and thus used) this suffices.

**Review by alcaeus — APPROVED** [review]
LGTM with a cautionary note about reusing tests ;)

File: tests/UnifiedSpecTests/UnifiedTestRunnerTest.php (hunk 2 above)
Comment (GromNaN) [reply tests/UnifiedSpecTests/UnifiedTestRunnerTest.php:11]: Indeed, I'll create a file specifically test the tester.

Comment (GromNaN) [reply tests/UnifiedSpecTests/UnifiedTestRunnerTest.php]: File added.

**Review by alcaeus — APPROVED** [review]

**Review by jmikola — APPROVED** [review]

**Pull request merged**